airflow/utils/db.py

```diff
@@ -655,16 +655,34 @@ def create_default_connections(session: Session = NEW_SESSION):
 @provide_session
 def initdb(session: Session = NEW_SESSION):
     """Initialize Airflow database."""
-    upgradedb(session=session)
-
+    from alembic import command
+    from flask import Flask
+    from flask_sqlalchemy import SQLAlchemy
+
+    from airflow.models import Base
+    from airflow.www.fab_security.sqla.models import Model
+    from airflow.www.session import AirflowDatabaseSessionInterface
+
+    def _create_flask_session_tbl():
+        flask_app = Flask(__name__)
+        flask_app.config['SQLALCHEMY_DATABASE_URI'] = conf.get('database', 'SQL_ALCHEMY_CONN')
+        db = SQLAlchemy(flask_app)
+        AirflowDatabaseSessionInterface(app=flask_app, db=db, table='session', key_prefix='')
+        db.create_all()
+
+    Base.metadata.create_all(settings.engine)
+    Model.metadata.create_all(settings.engine)
+    _create_flask_session_tbl()
+    # stamp the migration head
+    config = _get_alembic_config()
+    command.stamp(config, "head")
+    # Load default connections
     if conf.getboolean('database', 'LOAD_DEFAULT_CONNECTIONS'):
         create_default_connections(session=session)
-
-    with create_global_lock(session=session, lock=DBLocks.MIGRATIONS):
```

Comment:
Oh, when `airflow db init` is called this lock is no longer held. It probably should be.